service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionService.java

```diff
@@ -33,31 +40,66 @@ public CaseDataExtractionService(DateFormatterService dateFormatterService,
     }
 
     public Map<String, String> getNoticeOfProceedingTemplateData(CaseData caseData) {
-        HearingBooking hearingBooking = hearingBookingService.getMostUrgentHearingBooking(caseData);
+        Map<String, String> hearingBookingData = getHearingBookingData(caseData.getHearingDetails());
 
         // Validation within our frontend ensures that the following data is present
-        return Map.of(
-            "courtName", hmctsCourtLookupConfiguration.getCourt(caseData.getCaseLocalAuthority()).getName(),
-            "familyManCaseNumber", caseData.getFamilyManCaseNumber(),
-            "todaysDate", dateFormatterService.formatLocalDateToString(LocalDate.now(), FormatStyle.LONG),
-            "applicantName", getFirstApplicantName(caseData),
-            "orderTypes", getOrderTypes(caseData),
-            "childrenNames", getAllChildrenNames(caseData),
-            "hearingDate", dateFormatterService.formatLocalDateToString(hearingBooking.getDate(), FormatStyle.LONG),
-            "hearingVenue", hearingBooking.getVenue(),
-            "preHearingAttendance", hearingBooking.getPreHearingAttendance(),
-            "hearingTime", hearingBooking.getTime()
+        return ImmutableMap.<String, String>builder()
+            .put("courtName", getCourtName(caseData.getCaseLocalAuthority()))
+            .put("familyManCaseNumber", caseData.getFamilyManCaseNumber())
+            .put("todaysDate", dateFormatterService.formatLocalDateToString(LocalDate.now(), FormatStyle.LONG))
+            .put("applicantName", getFirstApplicantName(caseData.getApplicants()))
+            .put("orderTypes", getOrderTypes(caseData.getOrders()))
+            .put("childrenNames", getAllChildrenNames(caseData.getAllChildren()))
+            .put("judgeTitleAndName", formatJudgeTitleAndName(caseData.getJudgeAndLegalAdvisor()))
+            .put("legalAdvisorName", getLegalAdvisorName(caseData.getJudgeAndLegalAdvisor()))
+            .putAll(hearingBookingData)
+            .build();
+    }
+
+    private String getCourtName(String courtName) {
+        return hmctsCourtLookupConfiguration.getCourt(courtName).getName();
+    }
+
+    private String getLegalAdvisorName(JudgeAndLegalAdvisor judgeAndLegalAdvisor) {
+        if (judgeAndLegalAdvisor == null) {
```

Comment:
defaultIfNull would be better here

Reply:
I think these are fine here, this if statement is simply guarding against null pointer on '        return defaultIfNull(judgeAndLegalAdvisor.getLegalAdvisorName(), "");'